From: Juergen Beisert <j.beisert@pengutronix.de>
Subject: Fix shared libraries link paths

When cross compiling the install stage creates links that point to absolute
host paths:

$ ls -l <install_dir>
-rw-r--r--  1 jb users   182194 28. Jan 13:44 libblkid.a
lrwxrwxrwx  1 jb users       15 28. Jan 13:44 libblkid.so -> /usr/lib/libblkid.so.1.0*
lrwxrwxrwx  1 jb users       15 28. Jan 13:44 libblkid.so.1 -> libblkid.so.1.0*
-rwxr-xr-x  1 jb users   141967 28. Jan 13:44 libblkid.so.1.0*

With this patch applied:

-rw-r--r--  1 jb users   182194 28. Jan 13:44 libblkid.a
lrwxrwxrwx  1 jb users       15 28. Jan 13:44 libblkid.so -> libblkid.so.1.0*
lrwxrwxrwx  1 jb users       15 28. Jan 13:44 libblkid.so.1 -> libblkid.so.1.0*
-rwxr-xr-x  1 jb users   141967 28. Jan 13:44 libblkid.so.1.0*

---
 lib/Makefile.elf-lib |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

Index: lib/Makefile.elf-lib
===================================================================
--- lib/Makefile.elf-lib.orig
+++ lib/Makefile.elf-lib
@@ -45,8 +45,7 @@ install-shlibs install:: $(ELF_LIB) inst
 	@echo "	SYMLINK $(ELF_INSTALL_DIR)/$(ELF_SONAME)"
 	@$(LN_S) -f $(ELF_LIB) $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME)
 	@echo "	SYMLINK $(libdir)/$(ELF_IMAGE).so"
-	@$(LN_S) -f $(ELF_INSTALL_DIR)/$(ELF_SONAME) \
-		$(DESTDIR)$(libdir)/$(ELF_IMAGE).so
+	@$(LN_S) -f $(ELF_LIB) $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_IMAGE).so
 	@echo "	LDCONFIG"
 	@-$(LDCONFIG)
 
@@ -63,7 +62,7 @@ install-shlibs-strip: install-shlibs
 uninstall-shlibs uninstall::
 	$(RM) -f $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_LIB) \
 		$(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME) \
-		$(DESTDIR)$(libdir)/$(ELF_IMAGE).so
+		$(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_IMAGE).so
 	-$(LDCONFIG)
 
 clean::
